--- src/decompiled/Terraria/MessageBuffer.cs
+++ src/Terraria/Terraria/MessageBuffer.cs
@@ -1948,7 +_,7 @@
 						if (false)
 							Main.tile[num90, num91] = new Tile();
 
-						lock (Main.tile[num90, num91]) {
+						lock (Main.tile) { // TODO: See if stuff breaks
 							Main.tile[num90, num91].liquid = liquid;
 							Main.tile[num90, num91].liquidType(liquidType);
 							if (Main.netMode == 2)
